src/urh/signalprocessing/Signal.py

```diff
@@ -131,6 +135,27 @@ def __load_wav_file(self, filename: str):
 
         self.sample_rate = sample_rate
 
+    def __load_sub_file(self, filename: str):
+        # Flipper RAW file format (OOK): space separated values, number of samples above (positive value -> 1)
+        # or below (negative value -> 0) center
+        params = {"min": 0, "max": 255, "fmt": np.uint8}
+        params["center"] = (params["min"] + params["max"]) / 2
+        arr = []
+        with open(filename, 'r') as subfile:
+            for line in subfile:
+                dataline = re.match(r'RAW_Data:\s*([-0-9 ]+)\s*$', line)
+                if dataline:
+                    values = dataline[1].split(r' ')
+                    for value in values:
+                        intval = int(value)
+                        if intval > 0:
+                            arr.extend(np.full(intval, params["max"], dtype=params["fmt"]))
+                        else:
+                            arr.extend(np.zeros(-intval, dtype=params["fmt"]))
```

Comment:
Curious about the performance of this: As `arr` is a python list it needs to be converted to a numpy array down the road. Did we test loading larger sub files? Does it have an acceptable performance? An improvement would be to allocate a numpy array of desired size before entering the loop, e.g., with `arr = np.zeros(N)` and then set its values in the loop.

Reply:
Usually .sub files are rather small, so that wouldn't be a problem in practice. Due to the file format concept, you unfortunately do not know the number of samples beforehand. I would rather like to stick to the 1-pass solution and risk a slower performance for unusual large files.